src/Servers/Kestrel/Core/src/Internal/Http2/Http2FrameWriter.cs

```diff
@@ -214,6 +217,82 @@ public ValueTask<FlushResult> WriteResponseTrailers(int streamId, HttpResponseTr
             }
         }
 
+        private static int EncodeStatusCode(int statusCode, Span<byte> buffer)
+        {
+            switch (statusCode)
+            {
+                // Status codes which exist in the HTTP/2 StaticTable.
+                case 200:
+                case 204:
+                case 206:
+                case 304:
+                case 400:
+                case 404:
+                case 500:
+                    buffer[0] = (byte)(0x80 | H2StaticTable.StatusIndex[statusCode]);
```

Comment:
Encoding status codes should be done in H2StaticTable or HPackEncoder.

This is the same layering issue the client has, leaking the HPack implementation out in the caller.

Reply:
Yeah I'm going to change this to at least use a static helper method on HPackEncoder.

Reply:
DONE